ModuleNotFoundError for Numpy when pytest.approx fails

<!--
Thanks for submitting an issue!

Quick check-list while reporting bugs:
-->

### Description of the bug:
When using `pytest.approx` to compare lists in a test, a `ModuleNotFoundError` is raised for Numpy whenever the test fails. If the test succeeds, there is no such error.

It appears that `pytest.approx` does _not_ require Numpy to run, but the error message requires Numpy to display the details. Since `pytest.approx` does not require Numpy to function correctly, it should not require Numpy to display the error.

### Output of `pip list` from the virtual environment:
```
Package    Version
---------- -------
attrs      21.4.0
iniconfig  1.1.1
packaging  21.3
pip        22.0.3
pluggy     1.0.0
py         1.11.0
pyparsing  3.0.7
pytest     7.0.1
setuptools 49.2.1
tomli      2.0.1
```

### pytest and operating system versions:
```
$ python --version
Python 3.9.0
$ python -m pytest --version
pytest 7.0.1
```
macOS Big Sur

Version 11.6.2

### Minimal example:
```python
import pytest
def test_approx():
    assert [1, 2] == pytest.approx([1.001, 2.002])
```
#### Actual Result:
```
$ pytest
============================= test session starts ==============================
platform darwin -- Python 3.9.0, pytest-7.0.1, pluggy-1.0.0
rootdir: ****
collected 1 item                                                               

test_approx.py F                                                         [100%]

=================================== FAILURES ===================================
_________________________________ test_approx __________________________________

    def test_approx():
>       assert [1, 2] == pytest.approx([1.001, 2.002])
E       AssertionError: assert [1, 2] == approx([1.001...02 ± 2.0e-06])
E         (pytest_assertion plugin: representation of details failed: /Users/adalessa/Downloads/diffusion-master 2/venv/lib/python3.9/site-packages/_pytest/python_api.py:323: ModuleNotFoundError: No module named 'numpy'.
E          Probably an object has a faulty __repr__.)

test_approx.py:5: AssertionError
=========================== short test summary info ============================
FAILED test_approx.py::test_approx - AssertionError: assert [1, 2] == approx(...
============================== 1 failed in 0.04s ===============================
```
#### Expected result:
No `ModuleNotFoundError: No module named 'numpy'.` which makes the whole error message confusing and leads you to believe it failed because Numpy is not installed instead of the fact it was an assertion error.

--- a/AUTHORS
+++ b/AUTHORS
@@ -185,6 +185,7 @@ Katerina Koukiou
 Keri Volans
 Kevin Cox
 Kevin J. Foley
+Kian Eliasi
 Kian-Meng Ang
 Kodi B. Arfer
 Kojo Idrissa

--- a/changelog/9726.bugfix.rst
+++ b/changelog/9726.bugfix.rst
@@ -1,0 +1,2 @@
+Unnecessary numpy import in `ApproxSequenceLike` is removed.
+Also Approx's `test_error_messages` is separated for numpy and native datatypes.

--- a/changelog/9726.bugfix.rst
+++ b/changelog/9726.bugfix.rst
@@ -1,2 +1,1 @@
-Unnecessary numpy import in `ApproxSequenceLike` is removed.
-Also Approx's `test_error_messages` is separated for numpy and native datatypes.
+An unnecessary ``numpy`` import in `ApproxSequenceLike` was removed.

--- a/changelog/9726.bugfix.rst
+++ b/changelog/9726.bugfix.rst
@@ -1,1 +1,1 @@
-An unnecessary ``numpy`` import in `ApproxSequenceLike` was removed.
+An unnecessary ``numpy`` import inside :func:`pytest.approx` was removed.

--- a/src/_pytest/python_api.py
+++ b/src/_pytest/python_api.py
@@ -319,7 +319,6 @@ class ApproxSequenceLike(ApproxBase):
 
     def _repr_compare(self, other_side: Sequence[float]) -> List[str]:
         import math
-        import numpy as np
 
         if len(self.expected) != len(other_side):
             return [
@@ -340,7 +339,7 @@ class ApproxSequenceLike(ApproxBase):
                 abs_diff = abs(approx_value.expected - other_value)
                 max_abs_diff = max(max_abs_diff, abs_diff)
                 if other_value == 0.0:
-                    max_rel_diff = np.inf
+                    max_rel_diff = math.inf
                 else:
                     max_rel_diff = max(max_rel_diff, abs_diff / abs(other_value))
                 different_ids.append(i)

--- a/testing/python/approx.py
+++ b/testing/python/approx.py
@@ -92,9 +92,7 @@ SOME_INT = r"[0-9]+\s*"
 
 
 class TestApprox:
-    def test_error_messages(self, assert_approx_raises_regex):
-        np = pytest.importorskip("numpy")
-
+    def test_error_messages_native_dtypes(self, assert_approx_raises_regex):
         assert_approx_raises_regex(
             2.0,
             1.0,
@@ -134,6 +132,22 @@ class TestApprox:
                 rf"  3     \| {SOME_FLOAT} \| {SOME_FLOAT} ± {SOME_FLOAT}",
             ],
         )
+
+        # Specific test for comparison with 0.0 (relative diff will be 'inf')
+        assert_approx_raises_regex(
+            [0.0],
+            [1.0],
+            [
+                r"  comparison failed. Mismatched elements: 1 / 1:",
+                rf"  Max absolute difference: {SOME_FLOAT}",
+                r"  Max relative difference: inf",
+                r"  Index \| Obtained\s+\| Expected   ",
+                rf"\s*0\s*\| {SOME_FLOAT} \| {SOME_FLOAT} ± {SOME_FLOAT}",
+            ],
+        )
+
+    def test_error_messages_numpy_dtypes(self, assert_approx_raises_regex):
+        np = pytest.importorskip("numpy")
 
         a = np.linspace(0, 100, 20)
         b = np.linspace(0, 100, 20)
@@ -175,18 +189,6 @@ class TestApprox:
         )
 
         # Specific test for comparison with 0.0 (relative diff will be 'inf')
-        assert_approx_raises_regex(
-            [0.0],
-            [1.0],
-            [
-                r"  comparison failed. Mismatched elements: 1 / 1:",
-                rf"  Max absolute difference: {SOME_FLOAT}",
-                r"  Max relative difference: inf",
-                r"  Index \| Obtained\s+\| Expected   ",
-                rf"\s*0\s*\| {SOME_FLOAT} \| {SOME_FLOAT} ± {SOME_FLOAT}",
-            ],
-        )
-
         assert_approx_raises_regex(
             np.array([0.0]),
             np.array([1.0]),

